src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,294 @@
+     * @throws IllegalArgumentException If any of the given values is invalid.
+     */
+    public Book(final String genre, final String author, final String image,
+        final String title, final String subtitle, final String publisher,
+        final String year, final Integer pages, final String isbn) {
+        assertAuthor(author);
+        assertImage(image);
+        assertTitle(title);
+        assertSubtitle(subtitle);
+        assertPublisher(publisher);
+        assertYear(year);
+        assertPages(pages);
+        assertIsbn(isbn);
+        this.id = 0; // Will be set when saving by JPA provider
+        this.genre = genre;
+        this.author = author;
+        this.image = image;
+        this.title = title;
+        this.subtitle = subtitle;
+        this.publisher = publisher;
+        this.year = year;
+        this.pages = pages;
+        this.isbn = isbn;
+    }
+
+
+    /**
+     * @return The book's id.
+     */
+    public long getId() {
+        return id;
+    }
+
+    /**
+     * @return The book's genre.
+     */
+    public String getGenre() {
+        return genre;
+    }
+
+    /**
+     * @return The book's author.
+     */
+    public String getAuthor() {
+        return author;
+    }
+
+    /**
+     * @return The book's image.
+     */
+    public String getImage() {
+        return image;
+    }
+
+    /**
+     * @return The book's title.
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    /**
+     * @return The book's subtitle.
+     */
+    public String getSubtitle() {
+        return subtitle;
+    }
+
+    /**
+     * @return The book's publisher.
+     */
+    public String getPublisher() {
+        return publisher;
+    }
+
+    /**
+     * @return The year the book was/is/will be published.
+     */
+    public String getYear() {
+        return year;
+    }
+
+    /**
+     * @return The amount of pages in the book.
+     */
+    public int getPages() {
+        return pages;
+    }
+
+    /**
+     * @return The book's ISBN.
+     */
+    public String getIsbn() {
+        return isbn;
+    }
+
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Book)) {
+            return false;
+        }
+        final Book book = (Book) o;
+        return id == book.id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    @Override
+    public String toString() {
+        return "Book{" +
+            "id=" + id +
+            ", genre='" + genre + '\'' +
+            ", author='" + author + '\'' +
+            ", image='" + image + '\'' +
+            ", title='" + title + '\'' +
+            ", subtitle='" + subtitle + '\'' +
+            ", publisher='" + publisher + '\'' +
+            ", year='" + year + '\'' +
+            ", pages=" + pages +
+            ", isbn='" + isbn + '\'' +
+            '}';
+    }
```

Comment:
Maybe using [Lombok](https://projectlombok.org/features/Data) @Data reduce the boil of text of this class.

Reply:
Didn't know this project. I'll read about it now

Reply:
Okay, replaced stuff with lombok